service/biz/tpm_cert_verifier.go

```diff
@@ -96,11 +96,12 @@ func validateVerifyIakAndIDevIDCertsReq(req *VerifyIakAndIDevIDCertsReq) error {
 
 // getCertSerialNumber extracts the serial number from the cert subject serial number.
 func getCertSerialNumber(serial string) (string, error) {
-	// iakX509.Subject.SerialNumber comes in the format PID:xxxxxxx SN:1234JF
-	// Extract out the value after SN:
+	// iakX509.Subject.SerialNumber can come in the format PID:xxxxxxx SN:1234JF or just
+	// the serial number as is.
+	// Try to extract out the value after SN:
 	sn := strings.Split(serial, "SN:")
 	if len(sn) != 2 {
-		return "", fmt.Errorf("serial number %v is not in expected format", serial)
+		return sn[0], nil
```

Comment:
can sn[0] be ""? maybe add a non "" check here? and for sn[1] too if needed. And maybe add some test for this too such as the whole serial is a "".

Reply:
it compares the value to the serial number returned from the device, so if it's "", it'll fail there

Reply:
aha then maybe remove the error from (string, error)? as it doesn't really returning any error?